plasmapy/formulary/lengths.py

```diff
@@ -273,7 +273,13 @@ def _calculate_vperp_from_lorentzfactor(
         if relativistic and nans_in_both_T_and_Vperp:
             Vperp = np.copy(Vperp)
             rbody = RelativisticBody(particle, lorentz_factor=lorentzfactor)
-            Vperp[~isfinite_Vperp] = rbody.velocity
+            try:
+                Vperp[~isfinite_Vperp] = rbody.velocity
+            except ValueError:
+                # When Vperp is a scalar instead of an array, for example
+                # when a ParticleList is being provided
+                Vperp = rbody.velocity
+
```

Comment:
This works, but this should only be a temporary solution pending a more thorough refactoring of `gyroradius`.